content/blog/introduction-to-typescript/index.md

```diff
@@ -0,0 +1,212 @@
+
+However, this does mean that you can utilize the entire arsenal of Babel tooling to your disposal, [such as Babel plugins](https://babeljs.io/docs/en/plugins/).
+
+## Logic != Typings {#typings-are-not-logic}
+
+_TypeScript will not find all your typing errors on its own_. This is because TypeScript is only as useful as your typings are. [Let's look back at an earlier example](#type-safety):
+
+```javascript
+function addFive(input) {
+    return input + 5;
+}
+```
+
+If you keep `addFive`'s `input` parameter without an explicit type, it will try to do its best to detect the type based on the operations you run on the value. **While these inferred types are often very useful, it often has difficulty doing so in an accurate manor**. _This is why manually assigning types is almost always preferred to leaving them be inferred_. If it couldn't properly detect a type for the value in this example, it does nothing to prevent strings from being passed as the parameter value.
+
+Although examples like this are simple, strict typings can also become fairly complex ([see the Advanced section of the official handbook for such examples](https://www.typescriptlang.org/docs/handbook/advanced-types.html)) in order to maintain maximum type strictness.
+Also, because typings do nothing to test against the logic of your program, they should not be seen as a replacement for testing, but rather a companion to them. With strict typings and proper testing, regressions can be severely limited and improve code quality of life.
+
+##### Typing Mishaps Happen {#typings-can-be-wrong-too}
+
+Remember, because typings are kept separately from the project's logic code, typings can be misleading, incomplete, or otherwise incorrect. While this can also happen with TypeScript logic code, it tends to be more actively mitigated as a project's ability to compile (and therefore redistribute) relies on that typing information. This isn't to say that you should immediately mistrust typings, but simply a reminder that they too may have their flaws - just as any other codebase.
+
+## Don't Forget To Document {#typescript-is-not-documentation}
+
+Just as typings shouldn't replace tests, typings should also not replace documentation or comments. Typings can help understand what inputs and outputs you're expecting, but just the same as with testing; _it doesn't explain what the logic does or provide context as to why the data types have specific properties_, what the properties are used for, and so forth. Additionally, typings often do little to help explain how to contribute in the larger scale or the design principles that may be applied in the design of the application.
```

Comment:
> Additionally, typings often do little to help explain how to contribute in the larger scale or the design principles that may be applied in the design of the application.

I'm unclear on what this is talking about. I think minor rephrasing would help make this point land.

Reply:
```suggestion
Just as typings shouldn't replace tests, typings should also not replace documentation or comments. Typings can help understand what inputs and outputs you're expecting, but just the same as with testing; _it doesn't explain what the logic does or provide context as to why the data types have specific properties_, what the properties are used for, and so forth. Additionally, typings often do little to help explain how to contribute in the larger scale when talking about documentation. For example, in a large-scale application, there may be come complex data patterns or order-of-operations that are required to do a task. Typings alone will not effectively communicate these design principles that are integral to the usage of the code.
```